src/components/Paginator/Paginator.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { VFC, memo, useCallback } from 'react';
+import { Box } from '@chakra-ui/react';
+import { PaginatorInput } from './PaginatorInput';
+import { PaginatorButton } from './PaginatorButton';
+
+interface Props {
+
+  /** Page count. */
+  count: number;
+
+  /** Current page number. */
+  currentPage: number;
+
+  /** Change callback. */
+  onChange: (pageNumber: number) => void;
+
+  /**
+   * How many items would be visible from edges when close to edge,
+   * and how many items would be visible from sides of middle.
+   */
+  edgeCount?: number;
+}
+
+/**
+ * Paginator component.
+ */
+const PaginatorComponent: VFC<Props> = ({ count, currentPage, onChange, edgeCount = 2 }) => {
+  let leftButtons: number[] = [];
+  let middleButtons: number[] = [];
+  let rightButtons: number[] = [];
+
+  if (currentPage <= edgeCount + 2) {
+    leftButtons = Array.from({ length: currentPage + edgeCount }).map((_, i) => i + 1);
+    rightButtons = [count];
+  } else {
+    leftButtons = [1];
+
+    if (currentPage >= count - edgeCount - 1) {
+      rightButtons = Array.from({ length: count - currentPage + edgeCount + 1 }).map((_, i) => count - i)
+        .reverse();
+    } else {
+      middleButtons = Array.from({ length: 1 + edgeCount * 2 }).map((_, i) => i + (currentPage - edgeCount));
+      rightButtons = [count];
+    }
+  }
```

Comment:
Can we remake it as function that would have `[leftButtons, middleButtons, rightButtons]` as output?

Reply:
Let's always try to avoid `let` and any kind of mutations within component if possible